### PR TITLE
LOON-414: Leaderboard API

### PR DESCRIPTION
**Pull request by rposkocil**

### CODE REVIEW

File: backend/src/main/kotlin/cz/loono/backend/api/controller/LeaderboardController.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    private val leaderboardService: LeaderboardService
+) {
+
+    @GetMapping
+    fun getLeaderboard(
+        @RequestAttribute(name = Attributes.ATTR_BASIC_USER)
+        basicUser: BasicUser,
+        @RequestParam
+        size: Int = 100
+    ): LeaderboardDto {
```
Comment (martin-ladecky) [inline backend/src/main/kotlin/cz/loono/backend/api/controller/LeaderboardController.kt:26]: Som náročný, ale možno lepší názov pageSize?
Áno, zasa tá ukecaná java :).

File: backend/src/main/kotlin/cz/loono/backend/api/controller/LeaderboardController.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        return leaderboardService.getLeaderboard(basicUser.uid, size)
```
Comment (dzaru) [inline backend/src/main/kotlin/cz/loono/backend/api/controller/LeaderboardController.kt:27]: Expression?
```suggestion
    ): LeaderboardDto = leaderboardService.getLeaderboard(basicUser.uid, size)
```

File: backend/src/main/resources/doc/openapi.json [hunk 3]
```diff
@@ -1186,6 +1190,29 @@
                     "count"
                 ]
             },
+            "Badge": {
```
Comment (dzaru) [inline backend/src/main/resources/doc/openapi.json:1193]: Isn't it already in `master` branch, in other words, `git pull origin master`?

**Comment by dzaru** [comment]
Thanks for double quotes, mentioning the issue here #33

File: backend/src/main/kotlin/cz/loono/backend/api/controller/LeaderboardController.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    }
```
Comment (d35h) [inline backend/src/main/kotlin/cz/loono/backend/api/controller/LeaderboardController.kt:28]: Expression here?
```suggestion
    ): LeaderboardDto = leaderboardService.getLeaderboard(basicUser.uid, size)
```

**Comment by d35h** [comment]
Thanks for double quotes, mentioning it here #33

File: backend/src/main/kotlin/cz/loono/backend/api/controller/LeaderboardController.kt [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+class LeaderboardController(
+    private val leaderboardService: LeaderboardService
+) {
+
+    @GetMapping
+    fun getLeaderboard(
+        @RequestAttribute(name = Attributes.ATTR_BASIC_USER)
+        basicUser: BasicUser,
+        @RequestParam
+        size: Int = 100
```
[the remark this answers is not in the document]
Comment (rposkocil) [reply backend/src/main/kotlin/cz/loono/backend/api/controller/LeaderboardController.kt:25]: API nevi nic o page, dam tam leaderboardSize

**Review by d35h — APPROVED** [review]

**Review by martin-ladecky — APPROVED** [review]

**Pull request merged**